pandas/tests/test_join.py

```diff
@@ -348,3 +348,19 @@ def test_merge_join_categorical_multiindex():
     result = a.join(b, on=["Cat1", "Int1"])
     expected = expected.drop(["Cat", "Int"], axis=1)
     assert_frame_equal(expected, result)
+
+
+def test_left_index_and_right_index_true():
+    # From issue 28189
+
+    pdf = DataFrame({"idx": Categorical(["1"] * 4), "value": [1, 2, 3, 4]})
+    pdf = pdf.set_index("idx")
+    agg = pdf.groupby("idx").agg(np.sum)["value"]
```

Comment:
Just construct this directly - the less moving parts in a test the better

Reply:
Fixed the problems you pointed out.

Reply:
```suggestion
    df2 = pd.DataFrame([[6]], columns=["value"], index=pd.Index(pd.Categorical([1]), name="idx"))
```

Minor but the less moving parts the better, so would be good to get rid of the groupby stuff here

Reply:
Done!